### ISSUE TITLE
child_process: crashed while sending data

### ISSUE THREAD

**Issue by lolBig** [Rokid/ShadowNode#388]
```js
var isChild = process.env.IS_CHILD === 'true'

function ref() {
  setInterval(() => { }, 10000)
}

function runWithChild() {
  process.on('message', chunk => {
    console.log('on parent message', chunk)
  })
}

function runWithParent() {
  var path = require('path')
  var child_process = require('child_process')
  var selfPath = path.resolve(__dirname, __filename)
  var child = child_process.fork(selfPath, ['~~parent~~'], {
    env: {
      IS_CHILD: 'true',
    }
  })
  child.on('message', chunk => {
    console.log('on child message', chunk)
  })
  child.on('disconnect', () => {
    console.log('child disconnected')
  })
  setInterval(() => {
    child.send({
      cloud: false,
      intent: 'play_random',
      slots: {
        keyword: {
          pinyin: '',
          type: 'keyword',
          value: '歌'
        },
        want: {
          pinyin: '',
          type: 'want',
          value: '要'
        },
        me: {
          pinyin: '',
          type: 'me',
          value: '我'
        },
        play5: {
          pinyin: '',
          type: 'play5',
          value: '听'
        }
      },
      pattern: '^($trigger)?($please)?($you|$me)?($want)?($at)?($room)?($inside)?($play5)($other)?[($adj)($de)]?($keyword)($ah)?$',
      asr: '我要听歌',
      appId: 'xxxxxxxxxxxxxxxxxxx',
      appName: 'xxx音乐'
    })
  }, 30)
  console.log('forked child')
}

isChild ? runWithChild() : runWithParent()

ref()
```

after running the above code for a while, the child exit with below error:
```shell
TypeError: Invalid UTF-8 string
child disconnected
Error: ERR_IPC_CHANNEL_CLOSED
    at anonymous (/data/test.js:26:16)
```

this is caused by tcp multi packet, We should handle the case of a message in multiple packets in ipc

**Comment by lolBig** [comment]
fixed by #393 
closed

**Comment by yorkie** [comment]
Good works!
